src/component.jl

```diff
@@ -655,3 +655,38 @@ function _get_time_series_storage(component::InfrastructureSystemsComponent)
 
     return container.time_series_storage
 end
+
+function get_time_series_by_key(
+    key::TimeSeriesKey,
+    component::InfrastructureSystemsComponent;
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+    count::Union{Nothing, Int} = nothing,
+)
+    return get_time_series(
+        time_series_metadata_to_data(key.time_series_type),
+        component,
+        key.name,
+        start_time = start_time,
+        len = len,
+        count = count,
+    )
+end
+
+function get_time_series_by_key(
```

Comment:
Is this redundant with the previous function?

Reply:
This is for the case where we want to get the forecast for service cost, using TimerSeriesKey stored for variable cost.

Reply:
We might be missing each other.  The previous function accepts a key, which has `type` and `name`.  This function is identical except that it also takes a `name`.  It seems odd to me that you use the `name` parameter instead of using the `name` inside of the key.  Are the name values identical?  If so, then the function seems redundant.  If they are not the same then I think the interface is a bit confusing.